### PR TITLE
44 encapsulate the logic that reads energy profiles

### PR DESCRIPTION
**Pull request by Tklauw-TNO**
TODOs:
- [x] Fix tests that are currently failing
- [x] Build in support for loading price profiles on carriers (this should fix the electrolyzer test)
- [x] Make tests to check that profiles are correctly loaded, using a stub loader for testing the loading of profiles from influxDB

### CODE REVIEW

File: tests/test_profile_parsing.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    def __init__(self, energy_system: esdl.EnergySystem, file_path: Optional[Path]):
+        super().__init__(energy_system, file_path)
+        self._loaded_profiles = pd.read_csv(file_path, index_col="DateTime", parse_dates=True)
+
+    def _load_profile_timeseries_from_database(self, profile: esdl.InfluxDBProfile) -> pd.Series:
+        return self._loaded_profiles[profile.id]
+
+
+class TestProfileLoading(unittest.TestCase):
+    def test_loading_from_influx(self):
```
Comment (jimrojerTNO) [inline tests/test_profile_parsing.py:26]: Docstring?

File: tests/test_profile_parsing.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        for demand_name in ["HeatingDemand_2ab9", "HeatingDemand_6662", "HeatingDemand_506c"]:
+            profile_values = problem.get_timeseries(f"{demand_name}.target_heat_demand").values
+            self.assertEqual(profile_values[0], 0.0)
+            self.assertEqual(len(profile_values), 26)
+
+        heat_price_profile = problem.get_timeseries("Heat.price_profile").values
+        self.assertEqual(heat_price_profile[0], 0.0)
+        self.assertLess(max(heat_price_profile), 1.0)
+
+    def test_loading_from_csv(self):
```
Comment (jimrojerTNO) [inline tests/test_profile_parsing.py:53]: docstring

File: tests/test_profile_parsing.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            problem.get_timeseries("WindPark_7f14.maximum_electricity_source").values,
+        )
+
+        expected_array = np.array([1.0] * 3)
+        np.testing.assert_equal(expected_array, problem.get_timeseries("elec.price_profile").values)
+
+        expected_array = np.array([1.0e6] * 3)
+        np.testing.assert_equal(expected_array, problem.get_timeseries("gas.price_profile").values)
+
+    def test_loading_from_xml(self):
```
Comment (jimrojerTNO) [inline tests/test_profile_parsing.py:83]: docstring

File: tests/test_profile_parsing.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            input_timeseries_file="timeseries.xml",
+        )
+        problem.pre()
+
+        expected_array = np.array([1.5e5] * 16 + [1.0e5] * 13 + [0.5e5] * 16)
+        np.testing.assert_equal(
+            expected_array, problem.get_timeseries("demand.target_heat_demand").values
+        )
+
+    def test_loading_from_csv_with_influx_profiles_given(self):
```
Comment (jimrojerTNO) [inline tests/test_profile_parsing.py:106]: docstring

File: tests/test_profile_parsing.py (hunk 1 above)
Comment (Tklauw-TNO) [reply tests/test_profile_parsing.py:26]: added

File: tests/test_profile_parsing.py (hunk 2 above)
Comment (Tklauw-TNO) [reply tests/test_profile_parsing.py:53]: added

File: tests/test_profile_parsing.py (hunk 4 above)
Comment (Tklauw-TNO) [reply tests/test_profile_parsing.py:106]: added

**Review by jimrojerTNO — APPROVED** [review]

**Pull request merged**